### PR TITLE
feat: O3-310: Add an application-writable config.json file in frontends/

### PR DESCRIPTION
**Pull request by jnsereko**
## Requirements
- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.

#### For changes to apps
- [x] My work conforms to the [**OpenMRS 3.0 Styleguide**](https://om.rs/styleguide).

#### If applicable
- [x] My work includes tests or is validated by existing tests.
- [x] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary

Added a simple button to push the `config.json` file and show a success or error message to the user respectively 
@vasharma05 @hadijahkyampeire @denniskigen

## Screenshots

https://github.com/openmrs/openmrs-esm-core/assets/58003327/293abeeb-664f-4334-80d4-95b57c21e132

### CODE REVIEW

File: packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx [hunk 1]
```diff
@@ -184,6 +188,41 @@ export const Configuration: React.FC<ConfigurationProps> = () => {
...
+                    saveConfig(tempConfig).then(
+                      (response) => {
+                        if (response.ok) {
+                          showToast({
+                            critical: true,
+                            title: t("saveConfig.json", "Save Config.json"),
+                            kind: "success",
+                            description: t(
+                              "config.jsonSavedSuccessfully",
+                              "config.json saved successfully"
```
Comment (ibacher) [inline packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx:205]: "Successfully saved configuration"

File: packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx [hunk 2]
```diff
@@ -184,6 +188,41 @@ export const Configuration: React.FC<ConfigurationProps> = () => {
...
+                  kind="primary"
+                  iconDescription="push local config"
+                  renderIcon={(props) => <Upload size={16} {...props} />}
+                  onClick={() => {
+                    saveConfig(tempConfig).then(
+                      (response) => {
+                        if (response.ok) {
+                          showToast({
+                            critical: true,
+                            title: t("saveConfig.json", "Save Config.json"),
```
Comment (ibacher) [inline packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx:201]: "Saved Configuration"

File: packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx [hunk 3]
```diff
@@ -184,6 +188,41 @@ export const Configuration: React.FC<ConfigurationProps> = () => {
...
+                              "config.json saved successfully"
+                            ),
+                          });
+                        }
+                      },
+                      (error) => {
+                        showNotification({
+                          title: t(
+                            "saveConfig.jsonError",
+                            "Error saving config.json"
```
Comment (ibacher) [inline packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx:214]: "Error saving configuration"

File: packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx [hunk 4]
```diff
@@ -184,6 +188,41 @@ export const Configuration: React.FC<ConfigurationProps> = () => {
                 >
                   {t("clearConfig", "Clear Local Config")}
                 </Button>
+                <Button
+                  kind="primary"
+                  iconDescription="push local config"
+                  renderIcon={(props) => <Upload size={16} {...props} />}
+                  onClick={() => {
```
Comment (ibacher) [inline packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx:195]: Once the user saves the configuration, how do we get that configuration to take effect?

Comment (jnsereko) [reply packages/apps/esm-implementer-tools-app/src/configuration/configuration.component.tsx]: i will create a ticket for ensuring that configurations are loaded

**Comment by jnsereko** [comment]
hey @vasharma05 @hadijahkyampeire, kindly review this

**Review by vasharma05 — APPROVED** [review]

**Comment by brandones** [comment]
Okay, I got this up to date. I also made a few changes to the copy. I also made it so the implementer tools panel is under error notifications so that the latter can be seen when they come up.

If anyone (@ibacher , @vasharma05 ) have issues with my changes I'll make a new PR. Very excited to close this one :)

Thank you for your work on this @jnsereko !

**Pull request merged**